lib/components/ClinicUserSelect.js

```diff
@@ -0,0 +1,162 @@
+    setTargetUser: React.PropTypes.func.isRequired
+  },
+
+  handleChange: function(e) {
+    this.props.setTargetUser(e);
+  },
+
+  handleClickNext: function(e) {
+    e.preventDefault();
+    if(this.props.targetId){
+      this.props.onUserChange(this.props.targetId);
+    }
+  },
+
+  valueRenderer: function(option) {
+    var user = _.get(this.props.allUsers, option.value);
+    var name = _.get(user, 'fullName');
+    var bday = _.get(user, ['patient', 'birthday'], '');
+    var mrn = _.get(user, ['patient', 'mrn'], '');
+
+    var dateString = sundial.translateMask(bday, 'YYYY-MM-DD', 'M/D/YYYY');
+
+    return (
+      <div className={styles.optionLabelWrapper}>
+        <div className={styles.optionLabelName}>
+          {name} {mrn ? 'MRN:'+mrn : ''}
+        </div>
+        <div className={styles.optionLabelBirthday}>
+          {dateString}
+        </div>
+      </div>
+    );
+  },
+
+  renderError: function() {
+    var self = this;
+    if(this.props.updateProfileErrorMessage && !this.props.updateProfileErrorDismissed){
+      return (
+        <div className={styles.error}>
+          {this.props.updateProfileErrorMessage}
+          <i className={styles.iconClose} onClick={this.props.dismissUpdateProfileError}></i>
+        </div>
+      );
+    }
+  },
+
+  renderSelector: function(){
+    var allUsers = this.props.allUsers;
+    var targets = this.props.targetUsersForUpload;
+    var sorted = _.sortBy(targets, function(targetId) {
+      var targetInfo = allUsers[targetId];
+      if (_.get(targetInfo, ['patient','isOtherPerson'])) {
+        return targetInfo.patient.fullName;
+      }
+      return targetInfo.fullName;
+    });
+
+    var selectorOpts = _.map(sorted, function(targetId) {
+      var targetInfo = allUsers[targetId];
+      var mrn = _.get(targetInfo, ['patient', 'mrn'], '') ?  ' ' +_.get(targetInfo, ['patient', 'mrn']) : '';
+      if (_.get(targetInfo, ['patient','isOtherPerson'])) {
+        return {value: targetId, label: targetInfo.patient.fullName};
+      }
+      return {value: targetId, label: targetInfo.fullName + mrn};
+    });
+
+    return (
+      <Select clearable={false}
+        name={'uploadTargetSelect'}
+        onChange={this.handleChange}
+        options={selectorOpts}
+        simpleValue={true}
+        value={this.props.targetId}
+        className={styles.Select}
+        placeholder={'Search'}
+        optionRenderer={this.valueRenderer}
+        valueRenderer={this.valueRenderer} />
+    );
+  },
+
+  renderButton: function() {
+    var classes = cx({
+      [styles.button]: true,
+      disabled: !this.props.targetId
+    });
+    var disabled = !this.props.targetId ? 'disabled' : '';
+    return (
+      <div className={classes} disabled={disabled} onClick={this.handleClickNext}>
+        Next
+      </div>
+    );
+  },
+
+  renderAddNew: function() {
+    var classes = cx({
+      [styles.addLink]: true,
+      [styles.valueSelected]: this.props.targetId
```

Comment:
Stylistically, I prefer wrapping the truthiness use of `this.props.targetId` here in a `Boolean`. How do you feel?